### PR TITLE
[test] Added integration test for FC without servers

### PR DESCRIPTION
**Pull request by avedmala**
## Summary, imperative, start upper case, don't end with a period

Previously added fix for the edge case where FC is using request based metadata and the servers are down, resulting in a NPE. This PR is just asserting that we're getting the correct `VeniceClientException` during the case

## How was this PR tested?

Added new integration tests to `AvroStoreClientEndToEndTest`
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java [hunk 1]
```diff
@@ -215,6 +218,171 @@ protected void runTest(
...
+            try {
+              genericFastVsonClient.batchGet(keys).get();
+              fail();
+            } catch (VeniceClientException e) {
+              assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            } catch (ExecutionException e) {
+              assertTrue(e.getCause().getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+            }
+          }
+        } else if (batchGetKeySize == recordCnt) {
```
Comment (avedmala) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:264]: I don't think the case where `batchGetKeySize==100` is ever run since we only do the case where `batchGetKeySize==2`. Should I remove this case or leave it in so once we enable for other sizes we have the code?

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java [hunk 2]
```diff
@@ -215,6 +218,171 @@ protected void runTest(
...
+    AvroGenericStoreClient<String, GenericRecord> genericFastClient =
+        getGenericFastClient(clientConfigBuilder, metricsRepositoryForGenericClient, true);
+
+    AvroGenericStoreClient<String, Object> genericFastVsonClient = null;
+    // Construct a Vson store client
+    genericFastVsonClient =
+        getGenericFastVsonClient(clientConfigBuilder.clone(), new MetricsRepository(), vsonThinClient, true);
+    try {
+      if (batchGet) {
+        // test batch get of size 2 (current default max)
```
Comment (xunyin8) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:237]: I believe we do have stream batch get available that no longer have this limitation. Perhaps we could use that API instead since we are deprecating this old batch get API.

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java (hunk 1 above)
Comment (xunyin8) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:264]: Can we double check with @m-nagarajan? Since we are actually trying to move away from the 2 key batch get (temporary implementation) and use the more generic batch get that actually performs the scatter gather.

Comment (avedmala) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:264]: Talked to Manoj and we decided to enable the `batchGetKeySize==100` case for the tests

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java (hunk 2 above)
Comment (avedmala) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:237]: Removed test for `batchGetKeySize==2` here, Manoj gave the okay

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java [hunk 3]
```diff
@@ -215,6 +218,111 @@ protected void runTest(
...
+        } catch (VeniceClientException | ExecutionException e) {
+          assertTrue(e.getMessage().endsWith("metadata is not ready, attempting to re-initialize"));
+        }
+      } else {
+        for (int i = 0; i < recordCnt; ++i) {
+          String key = keyPrefix + i;
+
+          try {
+            genericFastClient.get(key).get();
+            fail();
```
Comment (m-nagarajan) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:261]: Can we reuse the `runTest` and add `expectedExceptions` to `testFastClientWithoutServers`'s Test annotation rather than adding a whole new function and checking the exception inline?

**Review by m-nagarajan — CHANGES_REQUESTED** [review]

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java (hunk 2 above)
Comment (avedmala) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java]: Reused `runTest` and removed this method altogether

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java (hunk 3 above)
Comment (avedmala) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java]: Made the refactor to reuse the method

Comment (m-nagarajan) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java]: As discussed offline, please make changes like [this](https://github.com/linkedin/venice/blob/main/clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/TestValidateSchemaAndBuildDictMapperOutputReader.java#L19). You don't have to handle whether you are getting the exception separately and let `expectedExceptions` handle it.

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java [hunk 4]
```diff
@@ -1,21 +1,24 @@
 package com.linkedin.venice.fastclient;
 
-import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.*;
```
Comment (m-nagarajan) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/fastclient/AvroStoreClientEndToEndTest.java:3]: Remove wildcards.

**Review by m-nagarajan — APPROVED** [review]
LGTM, thanks Abhinav.
